app/gui2/src/components/GraphEditor/GraphNodes.vue

```diff
@@ -1,15 +1,96 @@
 <script setup lang="ts">
 import GraphNode from '@/components/GraphEditor/GraphNode.vue'
+import { SnapGrid } from '@/components/GraphEditor/snap'
 import { injectGraphNavigator } from '@/providers/graphNavigator'
 import { injectGraphSelection } from '@/providers/graphSelection'
 import { useGraphStore } from '@/stores/graph'
-import type { Vec2 } from '@/util/vec2'
+import { useApproach } from '@/util/animation'
+import { Rect } from '@/util/rect'
+import { Vec2 } from '@/util/vec2'
+import { iteratorFilter } from 'lib0/iterator'
+import { abs } from 'lib0/math'
 import type { ContentRange, ExprId } from 'shared/yjsModel'
+import { ref, watchEffect, type Ref, type WatchStopHandle } from 'vue'
 
 const graphStore = useGraphStore()
 const selection = injectGraphSelection(true)
 const navigator = injectGraphNavigator(true)
 
+class Drag {
+  static THRESHOLD = 15.0
```

Comment:
all numbers in JS are `double`s so the `.0` is unnecessary, but if we want to use `.0` to indicate that it's semantically a `double` i'm fine with that. would be another thing to add to the style guide though

Reply:
It's just a habit from strongly typed languages.